.github/workflows/ci.yml

```diff
@@ -289,4 +289,89 @@ jobs:
       - name: Generate lockfile with minimal direct versions
         # Create a lockfile with minimal direct versions for reproducible builds.
         # See advisory: https://blog.rust-lang.org/2023/08/29/committing-lockfiles/
-        run: cargo +nightly generate-lockfile -Z direct-minimal-versions
\ No newline at end of file
+        run: cargo +nightly generate-lockfile -Z direct-minimal-versions
```

Comment:
Fantastic work getting this functioning. Really glad you stuck through and made it happen!

Reply:
Oops I mean the second section that does the tests :)

Reply:
Ah _that_ one.. The fruits of sheer stubbornness.